crates/contract-interface/src/types/updates.rs

```diff
@@ -0,0 +1,72 @@
+use crate::types::primitives::AccountId;
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+type Sha256Digest = [u8; 32];
+
+/// A vector of proposed updates
+#[derive(
+    Debug,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Hash,
+    Serialize,
+    Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+)]
+#[cfg_attr(
+    all(feature = "abi", not(target_arch = "wasm32")),
+    derive(schemars::JsonSchema)
+)]
+pub struct ProposedUpdates(pub Vec<Update>);
+
+/// A proposed update
+#[derive(
+    Debug,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Hash,
+    Serialize,
+    Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+)]
+#[cfg_attr(
+    all(feature = "abi", not(target_arch = "wasm32")),
+    derive(schemars::JsonSchema)
+)]
+pub struct Update {
+    pub update_id: u64,
+    pub update_hash: UpdateHash,
+    pub votes: Vec<AccountId>,
+}
```

Comment:
this is the dto type for one of the entries of  `entries: IterableMap<UpdateId, UpdateEntry>` where

```
struct UpdateEntry {
    update: Update,
    votes: HashSet<AccountId>,
    bytes_used: u128,
}
```

I guess the new names are better, but I am worried that the reason for the dto type to be different from the original gets lost.